File: backend/internal/tools/filters.go
package tools

// ToolFilters defines the available filters for tool queries
type ToolFilters struct {
	Category  string  // Category slug to filter by
	Price     string  // free, freemium, paid
	MinRating float64 // Minimum average rating
	Platform  string  // web, mobile, api (searches in platforms text field)
	Sort      string  // top_rated, most_bookmarked, trending, newest
}

// SortOptions defines valid sort options
const (
	SortTopRated       = "top_rated"
	SortMostBookmarked = "most_bookmarked"
	SortTrending       = "trending"
	SortNewest         = "newest"
)

// ValidateSort returns a valid sort option, defaulting to top_rated
func ValidateSort(sort string) string {
	switch sort {
	case SortTopRated, SortMostBookmarked, SortTrending, SortNewest:
		return sort
	default:
		return SortTopRated
	}
}

// ValidatePrice returns a valid price filter or empty string
func ValidatePrice(price string) string {
	switch price {
	case "free", "freemium", "paid":
		return price
	default:
		return ""
	}
}

File: frontend/src/stores/filters.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useFiltersStore = defineStore('filters', () => {
  const query = ref('')
  const category = ref<string | null>(null)
  const price = ref<string | null>(null)
  const minRating = ref<number | null>(null)
  const platform = ref<string | null>(null)
  const sort = ref('top_rated')

  function setFilter(key: string, value: any) {
    switch (key) {
      case 'query':
        query.value = value
        break
      case 'category':
        category.value = value
        break
      case 'price':
        price.value = value
        break
      case 'minRating':
        minRating.value = value
        break
      case 'platform':
        platform.value = value
        break
      case 'sort':
        sort.value = value
        break
    }
  }

  function clearFilters() {
    query.value = ''
    category.value = null
    price.value = null
    minRating.value = null
    platform.value = null
    sort.value = 'top_rated'
  }

  return {
    query,
    category,
    price,
    minRating,
    platform,
    sort,
    setFilter,
    clearFilters,
  }
})
